### PR TITLE
:bug: awsmachinetemplates: Allow cloudInit.secureSecretsBackend to be defaulted

### LINKED ISSUE #2110: AWSMachineTemplate defaulting is being rejected by validation logic in the webhook
/kind bug

**What steps did you take and what happened:**
1.  Created a cluster using v1alpha2 controller CAPI v0.2.11 and CAPA v0.4.10, with a MachineDeployment
2. Updated the webhooks CRDs and controller to CAPI 0.3.10 and CAPA 0.6.2
3. Increased the MachineDeployment replica.

**what happened:**
1. MachineDeployment failed to scale.
2. CAPI controller logged error saying 
```
E1115 17:59:42.719696       1 machineset_controller.go:151] controllers/MachineSet "msg"="Failed to reconcile MachineSet" "error"="admission webhook \"validation.awsmachinetemplate.infrastructure.x-k8s.io\" denied the request: AWSMachineTemplate.Spec is immutable" "machineset"="worker-01eq216a1dtzhhh78pw2fxptjv-additional-node-pool-2-76d494ff75" "namespace"="01eq1tgb8p4mtnxgkj5dqqh0xp"
E1115 17:59:42.720238       1 controller.go:257] controller-runtime/controller "msg"="Reconciler error" "error"="admission webhook \"validation.awsmachinetemplate.infrastructure.x-k8s.io\" denied the request: AWSMachineTemplate.Spec is immutable" "controller"="machinedeployment" "name"="worker-01eq1zqskzjzxkp6c64813kyrx-additional-node-pool-1" "namespace"="01eq1tgb8p4mtnxgkj5dqqh0xp"
E1115 17:59:42.720270       1 controller.go:257] controller-runtime/controller "msg"="Reconciler error" "error"="admission webhook \"validation.awsmachinetemplate.infrastructure.x-k8s.io\" denied the request: AWSMachineTemplate.Spec is immutable" "controller"="machineset" "name"="worker-01eq216a1dtzhhh78pw2fxptjv-additional-node-pool-2-76d494ff75" "namespace"="01eq1tgb8p4mtnxgkj5dqqh0xp"
E1115 17:59:42.720241       1 machineset_controller.go:151] controllers/MachineSet "msg"="Failed to reconcile MachineSet" "error"="admission webhook \"validation.awsmachinetemplate.infrastructure.x-k8s.io\" denied the request: AWSMachineTemplate.Spec is immutable" "machineset"="worker-01eq1zqskzjzxkp6c64813kyrx-additional-node-pool-1-56459f5b8b" "namespace"="01eq1tgb8p4mtnxgkj5dqqh0xp"
```

**What did you expect to happen:**
1. MachineDeployment to scale.

**Anything else you would like to add:**

https://github.com/kubernetes-sigs/cluster-api-provider-aws/commit/6e4da055615983cab9a8e9eeaf8175d3d3d6cc91#diff-fda9b90e86f8d2a7973a35e905b710db6b52116a4361f5fed68e388d096f43adc
 Here the open api defaulting is trying to set the field to secrets-manager, and the validation is rejecting it

Admission webhook error
```
admission webhook \"validation.awsmachinetemplate.infrastructure.x-k8s.io\" denied the request: AWSMachineTemplate.Spec is immutable :  v1alpha3.AWSMachineTemplateSpec{\n  \tTemplate: v1alpha3.AWSMachineTemplateResource{\n  \t\tSpec: v1alpha3.AWSMachineSpec{\n  \t\t\t... // 15 identical fields\n  \t\t\tNetworkInterfaces:    nil,\n  \t\t\tUncompressedUserData: nil,\n  \t\t\tCloudInit: v1alpha3.CloudInit{\n  \t\t\t\tInsecureSkipSecretsManager: false,\n  \t\t\t\tSecretCount:                0,\n  \t\t\t\tSecretPrefix:               \"\",\n- \t\t\t\tSecureSecretsBackend:       \"secrets-manager\",\n+ \t\t\t\tSecureSecretsBackend:       \"\",\n  \t\t\t},\n  \t\t\tSpotMarketOptions: nil,\n  \t\t\tTenancy:           \"\",\n  \t\t},\n  \t},\n  }\n" "object"={"kind":"MachineSet","namespace":"01eq1tgb8p4mtnxgkj5dqqh0xp","name":"worker-01eq1zqskzjzxkp6c64813kyrx-additional-node-pool-2-7d79dd7d5","uid":"d9b2fc68-6a0f-45e3-a170-751ad42c097e","apiVersion":"cluster.x-k8s.io/v1alpha3","resourceVersion":"166581557"
```

**Environment:**

- Cluster-api-provider-aws version: 0.6.2
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by randomvariable**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2110

### CODE REVIEW

File: api/v1alpha3/awsmachinetemplate_webhook.go [hunk 1]
```diff
@@ -60,8 +61,40 @@ func (r *AWSMachineTemplate) ValidateCreate() error {
...
-	if !reflect.DeepEqual(r.Spec, oldAWSMachineTemplate.Spec) {
+
+	newAWSMachineTemplateSpec := newAWSMachineTemplateUnstructured["spec"].(map[string]interface{})
+	oldAWSMachineTemplateSpec := oldAWSMachineTemplateUnstructured["spec"].(map[string]interface{})
+
+	newAWSMachineSpec := newAWSMachineTemplateSpec["template"].(map[string]interface{})["spec"].(map[string]interface{})
+	oldAWSMachineSpec := oldAWSMachineTemplateSpec["template"].(map[string]interface{})["spec"].(map[string]interface{})
+
+	// Allow setting of cloudInit.secureSecretsBackend to "secrets-manager" only to handle v1alpha3 upgrade
+	if oldAWSMachineTemplate.Spec.Template.Spec.CloudInit.SecureSecretsBackend == "" && r.Spec.Template.Spec.CloudInit.SecureSecretsBackend == SecretBackendSecretsManager {
```
Comment (sedefsavas) [inline api/v1alpha3/awsmachinetemplate_webhook.go:86]: Why do we use unstructured here?
We can set `r.Spec.Template.Spec.CloudInit.SecureSecretsBackend = "" instead?

**Comment by sedefsavas** [comment]
LGTM 
Just have a question above.

File: api/v1alpha3/awsmachinetemplate_webhook.go (hunk 1 above)
Comment (randomvariable) [reply api/v1alpha3/awsmachinetemplate_webhook.go:66]: Not sure. Going to blame doing it late at night...
Fixed.

**Pull request merged**